File: backend/researchcenter/src/main/java/edu/cit/uy/researchcenter/features/repository/model/RepositoryUpdate.java
package edu.cit.uy.researchcenter.features.repository.model;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

@Data
@Entity
@Table(name = "repository_updates")
public class RepositoryUpdate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "repository_id", nullable = false)
    private Long repositoryId;

    @Column(name = "author_id", nullable = false)
    private Long authorId;

    @Column(name = "author_name")
    private String authorName;

    @Transient
    private String authorProfilePicture;

    @Transient
    private String authorRole;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Instant updatedAt;
}

File: mobile/ResearchCenter/app/src/main/java/com/example/researchcenter/shared/model/RepositoryUpdate.kt
package com.example.researchcenter.shared.model

data class RepositoryUpdate(
    val id: Long,
    val repositoryId: Long,
    val content: String,
    val authorId: Long,
    val authorName: String,
    val createdAt: String,
    val updatedAt: String? = null,
    val authorProfilePicture: String? = null,
    val authorRole: String? = null
)
